assets/src/scripts/components/hydrograph/tooltip.js

```diff
@@ -128,18 +130,58 @@ const createTooltipTextGroup = function (elem, {currentPoints, comparePoints, qu
 
     // Add new text labels
     const newTexts = texts.enter()
-        .append('div')
-            .attr('class', d => `${d.tsKey}-tooltip-text`);
+        .append('div');
+
+    // Find the with of the between the y-axis and margin and set the tooltip margin based on that number
+    const adjustMarginOfTooltips = function (elem) {
+        // set a base number of pixels to bump the tooltips away from y-axis and compensate for under reporting of
+        // margin width by layout selector on time series with single or double digits on y-axis
+        const baseMarginOffsetTextGroup = 27;
+        elem.call(link(function (elem, layout) {
```

Comment:
You shouldn't nest calls to link. The issue is that you may end up calling the function on DOM that no longer exists. This is nested because createTooltipTextGroup is "linked".